bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableConnection.java

```diff
@@ -137,13 +139,16 @@ protected AbstractBigtableConnection(Configuration conf, boolean managed, Execut
 
     this.session = new BigtableSession(options);
 
-    int defaultRpcCount = AsyncExecutor.MAX_INFLIGHT_RPCS_DEFAULT * options.getChannelCount();
-    int maxInflightRpcs = conf.getInt(MAX_INFLIGHT_RPCS_KEY, defaultRpcCount);
-    long maxMemory = conf.getLong(
-      BIGTABLE_BUFFERED_MUTATOR_MAX_MEMORY_KEY,
-      AsyncExecutor.ASYNC_MUTATOR_MAX_MEMORY_DEFAULT);
-
-    this.heapSizeManager = new HeapSizeManager(maxMemory, maxInflightRpcs);
+    synchronized (resourceLimitedLock) {
```

Comment:
Can you please put this in a synchronized static method rather than having a separate Lock?